src/runtime/virtcontainers/qemu.go

```diff
@@ -841,8 +863,15 @@ func (q *qemu) startSandbox(ctx context.Context, timeout int) error {
 				strErr += string(b)
 			}
 		}
-
 		q.Logger().WithError(err).Errorf("failed to launch qemu: %s", strErr)
+
+		// stop virtiofsd
+		if q.config.SharedFS == config.VirtioFS {
```

Comment:
Hi @egernst 

Is it better to call the stopVirtiofsd in stopSandbox function, thus it can cover all of the startSandbox failed case?

Reply:
Sure, but we return before calling it here?

Reply:
Made a change to handle the cleanup as a defer, following what is done else where in the function and limiting the amount of changes elsewhere in the flow (ie, the stopSandbox handling).  

I do think this could use refactoring, but I'd rather not rush that (but do want to fix the current bug).